rep.go

```diff
@@ -81,6 +93,242 @@ func (rep *repSocket) SetOption(name string, value interface{}) error {
 	return rep.sck.SetOption(name, value)
 }
 
+type repMsg struct {
+	conn *Conn
+	msg  Msg
+}
+
+type repReader struct {
+	ctx   context.Context
+	mu    sync.Mutex
```

Comment:
it would seem `mu` is only guarding `conns`.
could you make that more obvious (using the usual Go way):
```go
type repReader struct {
    ctx context.Context
    state *repState

    mu sync.Mutex
    conns []*Conn

    msgC chan repMsg
}
```